code42_on_poll_connector.py

```diff
@@ -101,30 +101,40 @@ def __init__(self, connector, client, state):
         self._state = state or {}
 
     def handle_on_poll(self, param, action_result):
-        param = self._adjust_date_parameters(param)
-        severities = self._connector.get_config().get("severity")
-        if severities:
-            severities = severities.replace(" ", "").split(",")
-        query = build_alerts_query(
-            param["start_time"], param.get("end_time"), severities=severities
-        )
-        alerts = self._get_alerts(param, query)
         source_id = param.get("container_id")
         if source_id:
+            # Ignore all other query params when polling for specific alert IDs
             alert_ids = source_id.split(",")
-            alerts = [alert for alert in alerts if alert["id"] in alert_ids]
+            alerts = self._get_alert_details(alert_ids)
+            artifact_count = None
+        else:
+            param = self._adjust_date_parameters(param)
+            artifact_count = param.get(
+                "artifact_count", DEFAULT_ARTIFACT_COUNT_FOR_POLL_NOW
+            )
+            severities = self._connector.get_config().get("severity")
+            if severities:
+                severities = severities.replace(" ", "").split(",")
+            query = build_alerts_query(
+                param["start_time"], param.get("end_time"), severities=severities
+            )
+            alerts = self._get_alerts(param, query)
+            alert_ids = [alert["id"] for alert in alerts]
+            alerts = self._get_alert_details(alert_ids)
 
-        details = {}
+        alert = None
```

Comment:
If the `alerts` list is empty, won't this make the `.get()` later raise an exception? Should this be an empty dict instead?

Reply:
Oops, thank you

Reply:
I will add a test for this too